xlive/H2MOD.cpp

```diff
@@ -36,23 +39,19 @@ bool b_Infection = false;
 bool b_Halo2Final = false;
 bool b_HeadHunter = false;
 
-extern int H2GetInstanceId();
-extern XUID xFakeXuid[4];
-std::unordered_map<int, int> object_to_variant;
-
+std::unordered_map<wchar_t*, bool&> GametypesMap
```

Comment:
In reference to https://github.com/PermaNulled/cartographer/pull/242#issuecomment-514374292 hes probably right, an enum would be cleaner, however...

What you want to stride for is not having a custom map, you should be thinking, if 2moro someone wanted to add a new game type in code how many different spots would they need to update.

It'd be very nice if this map was driven by all the classes that implement GameClientServerHandler (i.e., register each sub class of GameClientServerHandler in some map). Then embed the string constant and whether its enabled in those classes as well. That way adding a new game type just involves implementing GameClientServerHandler. This would be the polymorphic way of handling this

You can even improve a lot of the event code (like player spawn and player death), for example

Right now you have 

`	if (b_HeadHunter)
	{
		DatumIndex dead_player = DatumIndex(unit_datum_index);
		headHunterHandler->playerDeath->SetDeadPlayer(dead_player); // set this so we can spawn a skull on their position.
		headHunterHandler->playerDeath->execute();
	}`

but it really should be

`	for (auto handler : game_handlers) {
		if (handler.enabled()) {
			handler->playerDeath.SetDeadPlayer(dead_player);
			handler->playerDeath->execute();
		}
	}`